pkg/policy/execute.go

```diff
@@ -221,6 +237,53 @@ func (e *Executor) Execute(ctx context.Context, req *ExecuteRequest, policy *Pol
 	return &total, nil
 }
 
+func (e *Executor) createCheckResult(ctx context.Context, selector []string, policyExecution *state.PolicyExecution, q *Check, qr *QueryResult, err error) error {
+	if policyExecution == nil {
+		return nil
+	}
+	checkResult := &state.CheckResult{
+		ExecutionId:        policyExecution.Id,
+		ExecutionTimestamp: policyExecution.Timestamp,
+		Name:               qr.Name,
+		Selector:           strings.Join(append(selector, q.Name), "/"),
+		Description:        qr.Description,
+		Status:             statusFailed,
+	}
+	if err != nil {
+		checkResult.Status = statusError
+		checkResult.Error = err.Error()
+		return e.stateClient.CreateCheckResult(ctx, checkResult)
+	}
+	if qr.Passed {
+		checkResult.Status = statusPassed
+	}
+
+	rows := make([]map[string]interface{}, len(qr.Rows))
+	for i, r := range qr.Rows {
+		m := make(map[string]interface{})
+		if len(r.AdditionalData) > 0 {
+			m["data"] = r.AdditionalData
+		}
+		if len(r.Identifiers) > 0 {
+			m["identifiers"] = r.Identifiers
+		}
+		if r.Reason != "" {
+			m["reason"] = r.Reason
+		}
+		if r.Status != "" {
+			m["status"] = r.Status
+		}
+		rows[i] = internal.FlattenRow(m)
```

Comment:
This implementation adds a `data.<Attribute_Name>`  I am wondering if this will be confusing to users as they write their own policies.

Reply:
good catch, maybewe should save all data as following:
If row =- Accountid: X, Region: Y, ARN: Z, SomeField: Z 
-> map == {identifiers: {ARN: Z}, status: something, reason: something, arn: Z,  SomeField: Z, Region: Y, AccountId: X}